docs/examples/auto-instrumentation/README.rst

```diff
@@ -0,0 +1,142 @@
+Autoinstrumentation
```

Comment:
general question: do we need the readme to be this verbose and explicit?

We have the getting started guide, which illustrates how to do instrumentation from scratch, as well as use instrumentations. I wonder if it's not additional maintenance overhead to provide a whole separate file for examples such as manual instrumentation? Maybe just showcasing auto-instrumentation would be enough?